components/brave_wallet/browser/eth_transaction.h

```diff
@@ -32,7 +32,8 @@ class EthTransaction {
   bool operator==(const EthTransaction&) const;
 
   static absl::optional<EthTransaction> FromTxData(
-      const mojom::TxDataPtr& tx_data);
+      const mojom::TxDataPtr& tx_data,
+      bool strict = true);
```

Comment:
we need test case for strict=false

Reply:
We didn't have tests at all for FromTxData so I added them for strict=false and for strict=true
e963a4de2ce6977af9e87c5554b4f709e02d0821